first_hour_on_vwb/working_with_data_collections.ipynb

```diff
@@ -0,0 +1,603 @@
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "class CreateWorkspaceWidget(object):\n",
+    "    def __init__(self):\n",
+    "        self.label = widgets.Label(value='Please provide appropriate values in the input boxes.')\n",
+    "        self.warning = wu.WarningWidget('The workspace name provided will be shown in the data catalog once the workspace is converted to a data collection.').get()\n",
+    "        self.input_name = wu.TextInputWidget(\"<WORKSPACE_NAME>\",\"Workspace Name:\").get()\n",
+    "        self.input_description = wu.TextInputWidget(\"<DESCRIPTION>\",\"Description:\").get()\n",
+    "        self.input_workspace_id = wu.TextInputWidget(\"<WORKSPACE_ID>\",\"Workspace ID:\").get()\n",
+    "        self.output_workspace_id = widgets.Text()\n",
+    "        self.output_workspace_id.value = self.input_workspace_id.value\n",
+    "        self.button = wu.StyledButton('Create workspace','Click to create a new workspace','plus').get()\n",
+    "        self.button.on_click(self.create_workspace)\n",
+    "        self.output = widgets.Output()\n",
+    "        self.vb = widgets.VBox(\n",
+    "            children = [self.label, self.warning,\n",
+    "                        self.input_name, self.input_description,\n",
+    "                        self.input_workspace_id,\n",
+    "                        self.button, self.output],\n",
+    "            layout = wu.vbox_layout)\n",
+    "        \n",
+    "    def get_workspace_id(self):\n",
+    "        return self.input_workspace_id.value\n",
+    "\n",
+    "    def create_workspace(self,b):\n",
+    "        with self.output:\n",
+    "            commandList = [\n",
+    "                \"terra\", \"workspace\", \"create\",\n",
+    "                f\"--id={self.input_workspace_id.value.strip()}\",\n",
+    "                f\"--description=\\\"{self.input_description.value}\\\"\",\n",
+    "                f\"--name={self.input_name.value.strip()}\",\n",
+    "            ]\n",
+    "\n",
+    "            print('Running command to create workspace...')\n",
+    "            print('\\n'.join(commandList))\n",
+    "            print('')\n",
+    "            print(\"Your workspace will be ready in less than one minute...\")\n",
+    "            \n",
+    "            result = subprocess.run(commandList, capture_output = True, text = True, timeout=180, check=True)\n",
+    "            print(result.stderr) if not result.stdout else print(result.stdout)\n",
+    "            \n",
+    "            # refreshWorkspaceContextCommand = \"terra workspace list\"\n",
+    "            # result = subprocess.run(refreshWorkspaceContextCommand, shell = True, capture_output = True, text = True, timeout=30, check=True)\n",
+    "\n",
+    "create_ws_widget = CreateWorkspaceWidget()\n",
+    "display(create_ws_widget.vb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
```

Comment:
Thinking about this further, a few comments:
- this is not quite as error-prone as I initially thought: even though the widget cell looks like it's done running ( it has no `[*]` indication), apparently the next cell won't run until it's finished.
- We do still need to worry about someone mindlessly running one cell after the other, since the dropdowns aren't set to the DC workspace
- So I think we could address this as mentioned in a subsequent comment: **let's set the focus in the subsequent dropdowns to the data collection workspace they just created**.  The notebook has that info, so hopefully it would be relatively straightforward to do so.  
That way, mindlessly clicking 'next' will do basically the right thing.

Reply:
In the initial version of this notebook, I did have each widget that follows the workspace creation one auto-populate with the newly created workspace's ID. However, it occurred to me that users might want to use the widgets out of order, in which case auto-population might be annoying/might prevent them from realizing they can use the conversion and version-publishing widgets for existing workspaces without creating a new WS. (You are usually the person who reminds me that users may not proceed in a linear fashion!)

It seems like it might work for both use cases (those who mindlessly run every cell in linear order AND those who want to use some subset of the widgets out of order)  to create a section of the notebook for converting existing workspaces to data collections and another section for publishing new versions of existing data collections. This will make the notebook substantially longer and essentially be duplicating existing code, for the most part, increasing the maintenance burden. What do you think?

Reply:
I think that's over-complicating :)
I'd say: if the new DC var is set, set the focus of the dropdown to that value; otherwise leave the focus unset (i.e., don't just set to the first in the list) so the user needs to make an explicit choice.